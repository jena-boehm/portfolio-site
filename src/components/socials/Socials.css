.socialsContainer {
        display: flex;
        justify-content: center;
        align-items: center;
        flex-direction: column;
        width: 100%;

    .icons {
        display: flex;
        justify-content: center;
        align-items: center;
        width: 100%;
        margin: 0 0 0 0.1em;

        .iconContainer {
            display: flex;
            justify-content: center;
            align-items: center;
            width: 17%;
            height: max-content;
            margin: 0 0.2em 0 0.2em;

            a { 
                display: flex;
            justify-content: center;
            align-items: center;
            }
        
            .iconBottom {
                width: 3%;
                position: absolute;
            }
            
            .iconBottom:active {
                transform: scale(1)
            }
        
            .iconTop {
                display: flex;
                justify-content: center;
                align-items: center;
                width: 100%;
                position: relative;
                opacity: 0;
                -moz-transition: all 0.2s ease-in-out 0s;
                -webkit-transition: all 0.2s ease-in-out 0s;
                -ms-transition: all 0.2s ease-in-out 0s;
                transition: all 0.2s ease-in-out 0s;
                cursor: pointer;

                .imgTop {
                width: 90%;
                }
            }

            .iconTop:hover {
                opacity: 1;
                transform: scale(1.1);
            }
                
            .iconTop:active {
                transform: scale(1)
            }
        }
    }

    button { 
        border: none;
        outline: none;
        background: transparent;
        text-transform: uppercase;
        /* box-shadow: 0 0 2px 0px #B0B6FB; */
        padding: 0.5em 1em 0.5em 1em;
        letter-spacing: 0.1em;
        cursor: pointer;
        font-size: 1em;
        color: #B0B6FB;
        transition: 0.2s;
        margin-top: 0.5em;
    }

    button:hover {
            color: white;
            transition: 0.3s;
    }

    button:active {
        transform: scale(0.9);    
    }
}