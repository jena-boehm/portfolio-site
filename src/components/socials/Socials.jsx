import React from 'react';
import styles from './Socials.css';
import linkedInBottom from '../../../public/icons/white-linkedin-icon.png';
import linkedInTop from '../../../public/icons/black-linkedin-icon.png';
import githubBottom from '../../../public/icons/white-github-icon.png';
import githubTop from '../../../public/icons/black-github-icon.png';
import mailBottom from '../../../public/icons/white-mail-icon.png';
import mailTop from '../../../public/icons/black-mail-icon.png';
import { resume, linkedin, github, email } from '../../data/data.js';

export const Socials = () => {

  const handleResume =  () => {
    window.open(resume);
  };
  
  return (
    <div className={styles.socialsContainer}>
      <div className={styles.icons}>
        <div className={styles.iconContainer}>
          <a href={linkedin} target="_blank" rel = "noopener noreferrer">
            <img 
              className={styles.iconBottom} 
              src={linkedInBottom} 
              alt="LinkedIn Icon" />
            <div className={styles.iconTop}>
              <img 
                className={styles.imgTop} 
                src={linkedInTop} 
                alt="LinkedIn Icon" />
            </div>
          </a>
        </div> 

        <div className={styles.iconContainer}>
          <a href={github} target="_blank" rel = "noopener noreferrer">
            <img 
              className={styles.iconBottom} 
              src={githubBottom} 
              alt="GitHub Icon" />
            <div className={styles.iconTop}>
              <img 
                className={styles.imgTop} 
                src={githubTop} 
                alt="GitHub Icon" />
            </div>
          </a>
        </div> 

        <div className={styles.iconContainer}>
          <a href={email} target="_blank" rel = "noopener noreferrer">
            <img 
              className={styles.iconBottom} 
              src={mailBottom} 
              alt="Email Icon" />
            <div className={styles.iconTop}>
              <img 
                className={styles.imgTop} 
                src={mailTop} 
                alt="Email Icon" />
            </div>
          </a>
        </div> 
      </div>
      <button onClick={handleResume}>Resume</button>
    </div>
  );
};
