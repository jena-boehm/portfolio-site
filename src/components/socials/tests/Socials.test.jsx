import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Socials } from '../Socials';

describe('Socials component', () => {
  afterEach(() => cleanup());
  it('renders Socials', () => {
    const { asFragment } = render(<Socials />);
    expect(asFragment()).toMatchSnapshot();
  });
});
