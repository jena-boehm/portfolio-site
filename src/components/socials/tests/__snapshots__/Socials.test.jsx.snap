// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Socials component renders Socials 1`] = `
<DocumentFragment>
  <div
    class="socialsContainer"
  >
    <div
      class="icons"
    >
      <div
        class="iconContainer"
      >
        <a
          href="https://www.linkedin.com/in/jenaluxboehm/"
        >
          <img
            alt="LinkedIn Icon"
            class="iconBottom"
            src="/path/image"
          />
          <div
            class="iconTop"
          >
            <img
              alt="LinkedIn Icon"
              class="imgTop"
              src="/path/image"
            />
          </div>
        </a>
      </div>
      <div
        class="iconContainer"
      >
        <a
          href="https://github.com/jena-boehm"
        >
          <img
            alt="GitHub Icon"
            class="iconBottom"
            src="/path/image"
          />
          <div
            class="iconTop"
          >
            <img
              alt="GitHub Icon"
              class="imgTop"
              src="/path/image"
            />
          </div>
        </a>
      </div>
      <div
        class="iconContainer"
      >
        <a
          href="mailto:jenaluxboehm@gmail.com"
        >
          <img
            alt="Email Icon"
            class="iconBottom"
            src="/path/image"
          />
          <div
            class="iconTop"
          >
            <img
              alt="Email Icon"
              class="imgTop"
              src="/path/image"
            />
          </div>
        </a>
      </div>
    </div>
    <button>
      Resume
    </button>
  </div>
</DocumentFragment>
`;
